api/v1alpha1/verticalscalingpolicy_types.go

```diff
@@ -153,3 +154,59 @@ func (status *VerticalScalingPolicyStatus) SetCondition(igName string, condition
 func (status *VerticalScalingPolicyStatus) SetConditions(igName string, conditions []*UtilizationCondition) {
 	status.TargetStatuses[igName].Conditions = conditions
 }
+
+func (vsp *VerticalScalingPolicy) Validate() error {
+	s := vsp.Spec
+
+	if s.Behavior == nil {
+		return errors.Errorf("validation failed, behavior not provided in spec")
+	}
+
+	if s.Behavior.ScaleUp != nil {
+		if s.Behavior.ScaleUp.Policies == nil {
+			return errors.Errorf("validation failed, no policies defined for scale up behavior in spec")
+		}
+
+		validPolicyTypes := []UtilizationType{CPUUtilizationPercent, MemoryUtilizationPercent, NodesCountUtilizationPercent}
+		for _, policy := range s.Behavior.ScaleUp.Policies {
+			if !isValidUtilizationType(validPolicyTypes, policy.Type) {
+				return errors.Errorf("validation failed, invalid policy type %s in scale up behaviors, valid types are: %s", policy.Type, validPolicyTypes)
+			}
+			if policy.Type == CPUUtilizationPercent || policy.Type == MemoryUtilizationPercent {
```

Comment:
How about something like `if strings.Contains("Percent", policy.Type)` instead? To avoid changing code in future when we have more policyTypes